### ISSUE TITLE
Bug related minuteInterval when date is also set

### ISSUE THREAD

**Issue by catalinbotean** [henninghall/react-native-date-picker#666]
**Description of the bug**
If interval is set (in this case is set to 5) and you open the date-picker with datetime mode and also the date is set and you choose the default option, the result is wrong. If you would choose any other time, it would work. See the attached video for more details.

https://github.com/henninghall/react-native-date-picker/assets/62332851/614b7877-f772-4beb-b009-39c0a24fc464

**Expected behavior**
If you choose the default date to not have the current time selected.

**To Reproduce**
```javascript
export default class App extends Component {

  render = () =>
    <DatePicker
      date={this.state.date}
      minuteInterval = {5}
      onDateChange={date => this.setState({ date })}
    />

}
```

**Smartphone (please complete the following information):**
 - OS: Android, iOS
 - React Native version 0.70.9
 - react-native-date-picker version 4.2.9

**Comment by vladanzlatar** [comment]
as a workaround you could round date by yourself:
```
const roundTime = (date: Date) => {
  const newDate = new Date(date.getTime());
  const minutes = newDate.getMinutes();
  const roundedMinutes = Math.floor(minutes / minuteInterval) * minuteInterval;
  newDate.setMinutes(roundedMinutes);
  newDate.setSeconds(0);
  return newDate;
};
```

**Comment by henninghall** [comment]
This bug has been fixed since version 4.2.12

**Comment by quangtuan28200** [comment]
> This bug has been fixed since version 4.2.12

same issue in 5.0.4
